docs/quickstart.rst

```diff
@@ -80,53 +83,58 @@ application that uses two tables that have a relationship to each other::
 
     class Post(db.Model):
         id = db.Column(db.Integer, primary_key=True)
-        title = db.Column(db.String(80))
-        body = db.Column(db.Text)
-        pub_date = db.Column(db.DateTime)
+        title = db.Column(db.String(80), nullable=False)
+        body = db.Column(db.Text, nullable=False)
+        pub_date = db.Column(db.DateTime, nullable=False,
+            default=datetime.utcnow)
 
-        category_id = db.Column(db.Integer, db.ForeignKey('category.id'))
+        category_id = db.Column(db.Integer, db.ForeignKey('category.id'),
+            nullable=False)
         category = db.relationship('Category',
-            backref=db.backref('posts', lazy='dynamic'))
-
-        def __init__(self, title, body, category, pub_date=None):
-            self.title = title
-            self.body = body
-            if pub_date is None:
-                pub_date = datetime.utcnow()
-            self.pub_date = pub_date
-            self.category = category
+            backref=db.backref('posts', lazy=True))
 
         def __repr__(self):
             return '<Post %r>' % self.title
 
 
     class Category(db.Model):
         id = db.Column(db.Integer, primary_key=True)
-        name = db.Column(db.String(50))
-
-        def __init__(self, name):
-            self.name = name
+        name = db.Column(db.String(50), nullable=False)
 
         def __repr__(self):
             return '<Category %r>' % self.name
 
 First let's create some objects::
 
-    >>> py = Category('Python')
-    >>> p = Post('Hello Python!', 'Python is pretty cool', py)
+    >>> py = Category(name='Python')
+    >>> Post(title='Hello Python!', body='Python is pretty cool', category=py)
+    >>> p = Post(title='Snakes', body='Ssssssss')
+    >>> py.posts.append(p)
     >>> db.session.add(py)
-    >>> db.session.add(p)
 
-Now because we declared ``posts`` as dynamic relationship in the backref
-it shows up as query::
+As you can see, there is no need to add the `Post` objects to the
+session. Since the `Category` is part of the session all objects
+associated with it through relationships will be added too.  It does
+not matter whether ``db.session.add()`` is called before or after
+creating these objects.  The association can also be done on either
+side of the relationship - so a post can be created with a category
+or it can be added to the list of posts of the category.
+
+Let's look at the posts. Accessing them will load them from the database
+since the relationship is lazy-loaded, but you will probably not notice
+the difference - loading a list is quite fast. Just don't do it when
+looping over many objects (in such a case you would rather use the
+``'subquery'`` or ``'joined'`` loading strategy in the model or as a
```

Comment:
Might be helpful to show an example of when and how to use the `joinedload` option.